runtime/builtin_types_test.go

```diff
@@ -425,6 +425,51 @@ func TestBuiltinSetAttr(t *testing.T) {
 	}
 }
 
+func TestRawInput(t *testing.T) {
+	fun := newBuiltinFunction("TestRawInput", func(f *Frame, args Args, _ KWArgs) (*Object, *BaseException) {
+		// Create a fake Stdin for input test.
+		stdinFile, w, err := os.Pipe()
+		if err != nil {
+			return nil, f.RaiseType(RuntimeErrorType, fmt.Sprintf("failed to open pipe: %v", err))
+		}
+
+		go func() {
+			w.Write([]byte("hello grumpy\n"))
+			w.Sync()
+		}()
+
+		oldStdin := Stdin
+		Stdin = NewFileFromFD(stdinFile.Fd())
+		defer func() {
+			Stdin = oldStdin
+			stdinFile.Close()
+		}()
+
+		var input *Object
+		output, raised := captureStdout(f, func() *BaseException {
+			in, raised := builtinRawInput(f, args, nil)
+			input = in
+			return raised
+		})
+
+		if raised != nil {
+			return nil, raised
+		}
+
+		return newTestTuple(input, output).ToObject(), nil
+	}).ToObject()
+	cases := []invokeTestCase{
+		{args: wrapArgs(), want: newTestTuple("hello grumpy", "").ToObject()},
+		{args: wrapArgs("abc"), want: newTestTuple("hello grumpy", "abc").ToObject()},
```

Comment:
I think there's a few more things that need testing:

- Multiline prompt parameter: make sure all string lines are printed
- Multiple lines on stdin: only the first line should be returned by the raw_input call
- An empty stdin should raise EOFError:

```
$ python -c 'raw_input()' < /dev/null
Traceback (most recent call last):
  File "<string>", line 1, in <module>
EOFError: EOF when reading a line
```

The last two test cases can be supported by having `fun` take a parameter that is the string to put on stdin.